packages/core/src/fhir-to-cda/cda-templates/components/encounters.ts

```diff
@@ -0,0 +1,296 @@
+const tableHeaders = [
+  "Reason for Visit",
+  "Encounter Type",
+  "Attending Provider",
+  "Location",
+  "Date",
+];
+
+export function buildEncounters(fhirBundle: Bundle): EncountersSection {
+  const encounters: Encounter[] =
+    fhirBundle.entry?.flatMap(entry => (isEncounter(entry.resource) ? [entry.resource] : [])) || [];
+
+  if (encounters.length === 0) {
+    return undefined;
+  }
+
+  const augmentedEncounters = createAugmentedEncounters(encounters, fhirBundle);
+
+  const { trs, entries } = createTableRowsAndEntries(
+    augmentedEncounters,
+    createTableRowFromEncounter,
+    createEntryFromEncounter
+  );
+
+  const table = initiateSectionTable(encountersSectionName, tableHeaders, trs);
+
+  return {
+    templateId: buildInstanceIdentifier({
+      root: oids.encountersSection,
+      extension: extensionValue2015,
+    }),
+    code: buildCodeCe({
+      code: "46240-8",
+      codeSystem: loincCodeSystem,
+      codeSystemName: loincSystemName,
+      displayName: "History of encounters",
+    }),
+    title: "ENCOUNTERS",
+    text: table,
+    entry: entries,
+  };
+}
+
+function createAugmentedEncounters(
+  encounters: Encounter[],
+  fhirBundle: Bundle
+): AugmentedEncounter[] {
+  const augEncs = encounters.map(encounter => {
+    const practitionerIds = encounter.participant?.flatMap(p =>
+      p.individual?.reference?.includes("Practitioner") ? p.individual?.reference : []
+    );
+
+    const involvedPractitioners: Practitioner[] = [];
+    practitionerIds?.forEach(id => {
+      const practitioner = findResourceInBundle(fhirBundle, id);
+      if (isPractitioner(practitioner)) involvedPractitioners.push(practitioner);
+    });
+
+    const locationIds = encounter.location?.flatMap(l =>
+      l.location?.reference?.includes("Location") ? l.location?.reference : []
+    );
+
+    const involvedLocations: Location[] = [];
+    locationIds?.forEach(id => {
+      const location = findResourceInBundle(fhirBundle, id);
+      if (isLocation(location)) involvedLocations.push(location);
+    });
+
+    return new AugmentedEncounter(
+      encountersSectionName,
+      encounter,
+      involvedPractitioners,
+      involvedLocations
+    );
+  });
+
+  return augEncs;
+}
+
+function createTableRowFromEncounter(
+  encounter: AugmentedEncounter,
+  referenceId: string
+): ObservationTableRow[] {
+  const locationInfo = getLocationInformation(encounter.locations);
+  return [
+    {
+      tr: {
+        _ID: referenceId,
+        ["td"]: [
+          {
+            "#text":
+              getDisplaysFromCodeableConcepts(encounter.resource.reasonCode) ?? NOT_SPECIFIED,
+          },
+          {
+            "#text": getDisplaysFromCodeableConcepts(encounter.resource.type) ?? NOT_SPECIFIED,
+          },
+          {
+            "#text": getPractitionerInformation(encounter.practitioners),
+          },
+          {
+            "#text": locationInfo?.map(l => `${l.name} - ${l.address}`).join("\n") ?? NOT_SPECIFIED,
```

Comment:
Do we want them to have new lines or the actual string literal "\n"?

Reply:
does that make a difference?

Reply:
yeah, I mean:

A - New lines in the XML:

```xml
  <text>first line
second line
third line</text>
```

B - New lines as part of the content

```xml
  <text>first line\nsecond line\nthird line</text>
```

B would help w/ formatting, A I don't know what it helps with.